physics/MP/Thompson/module_mp_thompson.F90

```diff
@@ -433,554 +431,554 @@ MODULE module_mp_thompson
 
 !>  - Compute evaporation of rain; ignore depositional growth of rain
-      t1_qr_ev = 0.78 * crg(10)
-      t2_qr_ev = 0.308*Sc3*SQRT(av_r) * crg(11)
+         t1_qr_ev = 0.78 * crg(10)
+         t2_qr_ev = 0.308*Sc3*SQRT(av_r) * crg(11)
 
 !>  - Compute sublimation/depositional growth of snow
-      t1_qs_sd = 0.86
-      t2_qs_sd = 0.28*Sc3*SQRT(av_s)
+         t1_qs_sd = 0.86
+         t2_qs_sd = 0.28*Sc3*SQRT(av_s)
 
 !>  - Compute melting of snow
-      t1_qs_me = PI*4.*C_sqrd*olfus * 0.86
-      t2_qs_me = PI*4.*C_sqrd*olfus * 0.28*Sc3*SQRT(av_s)
+         t1_qs_me = PI*4.*C_sqrd*olfus * 0.86
+         t2_qs_me = PI*4.*C_sqrd*olfus * 0.28*Sc3*SQRT(av_s)
 
 !>  - Compute sublimation/depositional growth of graupel
-      t1_qg_sd = 0.86 * cgg(10)
-      t2_qg_sd = 0.28*Sc3*SQRT(av_g) * cgg(11)
+         t1_qg_sd = 0.86 * cgg(10)
+         t2_qg_sd = 0.28*Sc3*SQRT(av_g) * cgg(11)
 
 !>  - Compute melting of graupel
-      t1_qg_me = PI*4.*C_cube*olfus * 0.86 * cgg(10)
-      t2_qg_me = PI*4.*C_cube*olfus * 0.28*Sc3*SQRT(av_g) * cgg(11)
+         t1_qg_me = PI*4.*C_cube*olfus * 0.86 * cgg(10)
+         t2_qg_me = PI*4.*C_cube*olfus * 0.28*Sc3*SQRT(av_g) * cgg(11)
 
 !>  - Compute constants for helping find lookup table indexes
-      nic2 = NINT(ALOG10(r_c(1)))
-      nii2 = NINT(ALOG10(r_i(1)))
-      nii3 = NINT(ALOG10(Nt_i(1)))
-      nir2 = NINT(ALOG10(r_r(1)))
-      nir3 = NINT(ALOG10(N0r_exp(1)))
-      nis2 = NINT(ALOG10(r_s(1)))
-      nig2 = NINT(ALOG10(r_g(1)))
-      nig3 = NINT(ALOG10(N0g_exp(1)))
-      niIN2 = NINT(ALOG10(Nt_IN(1)))
+         nic2 = nint(log10(r_c(1)))
+         nii2 = nint(log10(r_i(1)))
+         nii3 = nint(log10(Nt_i(1)))
+         nir2 = nint(log10(r_r(1)))
+         nir3 = nint(log10(N0r_exp(1)))
+         nis2 = nint(log10(r_s(1)))
+         nig2 = nint(log10(r_g(1)))
+         nig3 = nint(log10(N0g_exp(1)))
+         niIN2 = nint(log10(Nt_IN(1)))
 
 !>  - Create bins of cloud water (from min diameter up to 100 microns)
-      Dc(1) = D0c*1.0d0
-      dtc(1) = D0c*1.0d0
-      do n = 2, nbc
-         Dc(n) = Dc(n-1) + 1.0D-6
-         dtc(n) = (Dc(n) - Dc(n-1))
-      enddo
+         Dc(1) = D0c*1.0_dp
+         dtc(1) = D0c*1.0_dp
+         do n = 2, nbc
+            Dc(n) = Dc(n-1) + 1.e-6_dp
+            dtc(n) = (Dc(n) - Dc(n-1))
+         enddo
 
 !>  - Create bins of cloud ice (from min diameter up to 2x min snow size)
-      xDx(1) = D0i*1.0d0
-      xDx(nbi+1) = 2.0d0*D0s
-      do n = 2, nbi
-         xDx(n) = DEXP(DFLOAT(n-1)/DFLOAT(nbi) &
-                  *DLOG(xDx(nbi+1)/xDx(1)) +DLOG(xDx(1)))
-      enddo
-      do n = 1, nbi
-         Di(n) = DSQRT(xDx(n)*xDx(n+1))
-         dti(n) = xDx(n+1) - xDx(n)
-      enddo
+         xDx(1) = D0i*1.0_dp
+         xDx(nbi+1) = D0s*2.0_dp
+         do n = 2, nbi
+            xDx(n) = exp(real(n-1, kind=dp)/real(nbi, kind=dp) &
+                     *log(real(xDx(nbi+1)/xDx(1), kind=dp)) + log(real(xDx(1), kind=dp)))
```

Comment:
Is it really required to have `real(X,dp)` inside the logs when the argument is already the precision required? This same question obviously applies to many of the statements below too.

Reply:
@grantfirl Correct, integers only. xD is defined type dp already

Reply:
@grantfirl I fixed as many of these as I could find. See latest commit.